yarn-project/types/src/tx.ts

```diff
@@ -37,6 +37,9 @@ export class Tx {
   }
 
   public static createPublic(txRequest: SignedTxRequest): PublicTx {
+    // public transactions can't be constructors or private:
+    txRequest.txRequest.functionData.isConstructor = false;
+    txRequest.txRequest.functionData.isPrivate = false;
```

Comment:
Strong no from me on silently modifying the txRequest object being passed. We should check those values and throw if they're invalid, so we can fix them at the caller. However, if making this change now opens a can of worms, I'm fine converting it into an issue that we tackle afterwards.

Reply:
Right yeah, this seems pretty suspect. Why are these being modified here?

Reply:
hmm good point - I have moved this to be int he sequencer test then!